### PR TITLE
Fix bastion for multi subnet

### PR DESCRIPTION
**Pull request by kon-angelo**
**How to categorize this PR?**

/area control-plane
/kind enhancement
/platform azure

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```bug user
Bastion now correctly computes network information (vnet, subnet) from infrastructure status.
```
```feature user
Add bastion support for multi-subnet deployments
```

### CODE REVIEW

**Comment by tedteng** [comment]
@kon-angelo Should we process this PR?

File: pkg/controller/bastion/actuator_reconcile.go [hunk 1]
```diff
@@ -62,7 +70,15 @@ func (a *actuator) Reconcile(ctx context.Context, bastion *extensionsv1alpha1.Ba
 		return err
 	}
 
-	nic, err := ensureNic(ctx, factory, opt, publicIP)
+	if infrastructureStatus.Networks.Layout != "SingleSubnet" {
+		return fmt.Errorf("unsupported network layout %s", infrastructureStatus.Networks.Layout)
```
[the remark this answers is not in the document]
Comment (tedteng) [reply pkg/controller/bastion/actuator_reconcile.go:74]: I guess forget cherry-pick 042d16bd153e5af48243d0b339062a6a754c5569 @kon-angelo

[the remark this answers is not in the document]
Comment (kon-angelo) [reply pkg/controller/bastion/actuator_reconcile.go:74]: Yes indeed I did. I will push an update

File: pkg/controller/bastion/actuator_reconcile.go [hunk 2]
```diff
@@ -110,6 +126,28 @@ func (a *actuator) Reconcile(ctx context.Context, bastion *extensionsv1alpha1.Ba
 	return a.client.Status().Patch(ctx, bastion, patch)
 }
 
+func getInfrastructureStatus(ctx context.Context, a *actuator, cluster *extensions.Cluster) (*azure.InfrastructureStatus, error) {
+	var infrastructureStatus *azure.InfrastructureStatus
+	worker := &extensionsv1alpha1.Worker{}
+	err := a.client.Get(ctx, client.ObjectKey{Namespace: cluster.ObjectMeta.Name, Name: cluster.Shoot.Name}, worker)
+	if err != nil {
```
[the remark this answers is not in the document]
Comment (kon-angelo) [reply pkg/controller/bastion/actuator_reconcile.go:133]: I need to declare it for line 141 to work, but i will clean up the function a bit with next update

[the remark this answers is not in the document]
Comment (dkistner) [reply pkg/controller/bastion/actuator_reconcile.go:133]: ahh right my bad.

**Review by dkistner — APPROVED** [review]
/lgtm

**Pull request merged**